randovania/gui/lib/multiplayer_session_api.py

```diff
@@ -26,11 +28,11 @@
 OriginalFunc = typing.Callable[Param, RetType]
 
 
-def handle_network_errors[**Param, RetType](
-    fn: typing.Callable[typing.Concatenate[MultiplayerSessionApi, Param], RetType],
-) -> typing.Callable[Param, RetType]:
```

Comment:
hmm. seeing this concatenate has mii questioning if my signature was correct. maybe it should be:
```py
def handle_network_errors[T, **P](
    fn: Callable[Concatenate[MultiplayerSessionApi, P], Coroutine[Any, Any, T]],
) -> Callable[Concatenate[MultiplayerSessionApi, P], Coroutine[Any, Any, T | None]]:
    @functools.wraps(fn)
    async def wrapper(self: MultiplayerSessionApi, *args: P.args, **kwargs: P.kwargs) -> T | None:
```
I'm not sure if the change is necessary, tho

Reply:
I don't really see a difference with or without your suggestion.
In both cases they behave the same, in regards to mypy complaining, on the current code and if i plop the decorator on a function that doesnt have `self: MultiplayerSessionApi`